src/briefcase/platforms/linux/flatpak.py

```diff
@@ -149,6 +152,38 @@ def permissions_context(self, app: AppConfig, x_permissions: dict[str, str]):
             "finish_args": finish_args,
         }
 
+    def _write_requirements_file(
+        self, app: AppConfig, requires: list[str], requirements_path: Path
+    ):
+        super()._write_requirements_file(app, requires, requirements_path)
+
+        # Flatpak runs ``pip install`` using an ``install_requirements.sh`` which Briefcase uses
+        # to indicate user-configured arguments to the command
+
+        pip_install_command = " ".join(
+            [
+                "/app/bin/python3",
```

Comment:
Don't want to interject myself too much here....but potentially something to consider.

I've tried to be leery of coupling Briefcase too much to its first-party templates. For instance, here, we're assuming where Python will be available in the Flatpak; this will limit where a custom template can put Python without also re-implementing this packaging format for Briefcase. So (without having considered any potential complexity for this...), perhaps Python could be added to `PATH` in the Flatpak (or a similar mechanism) to prevent this coupling. freakboy3742, I'll ofc defer to you for any prioritization of this concern :)

p.s. - definitely appreciating your contributions :)

Reply:
Agreed that we should be avoiding coupling Briefcase to an "official" template wherever possible; but in this case, I think we're reasonable safe. `/app/bin` is where binaries will almost certainly be installed for virtual all practical Flatpak purposes; if that turns out to *not* be the case, we could add an extension point to the template that defines `flatpak_python_exe = "/app/bin/python3"` (similar to how we configure the location of the support package etc). However, that seems like a fairly niche requirement; and it's a fairly simple thing to add if and when the requirement emerges.

Reply:
> I've tried to be leery of coupling Briefcase too much to its first-party templates. For instance, here, we're assuming where Python will be available in the Flatpak

I suppose by this argument, we should also allow the template to control where the pip options are written, rather than assuming that install_requirements.sh is in the same directory as requirements.txt.

In fact, I have an idea which would solve both this and the Python executable path issue. In #1270 I suggested that the Android mechanism would be:

> make Briefcase write the pip options to a simple text file, one option per line, and put some code in the template's build.gradle to read the options from the file and pass them to the `pip { options }` method

What if Flatpak worked in the same way? 

* Briefcase would use the same code for both Flatpak and Android.
* This code would write only the options file, not a full shell script.
* The location and name of the file would be controlled by a variable in the template itself, just like [the requirements.txt file](https://github.com/beeware/briefcase-linux-flatpak-template/blob/v0.3.20/%7B%7B%20cookiecutter.format%20%7D%7D/briefcase.toml#L7).
* The details of how the file is used would be contained entirely within the template.

Reply:
@mhsmith so would the file to write have a generic name like the configuration option? I suppose I can see a version of this feature where the arguments are always written in the same format regardless of target (and therefore, it can be part of the base `CreateCommand` without private method overrides like I've done in this PR)... and then it's up to the template whether or how it uses those arguments.

On the other hand, keeping in mind that arbitrary requirement installer backends could be supported aside from pip (like conda, uv), writing the pip options to a file makes the requirement installer backend a template detail, rather than a Briefcase detail... but only for Flatpak and Android which run `pip install` themselves rather than other targets where Briefcase executes it.

Implementing another installer backend then gets split. For most targets, it would just be a change to Briefcase to support it. For targets that use the `pip-options` file, they would need to have logic in the template to use the correct installer backend, adding complexity to the template.

Of course for Flatpak more would need to be done to support some other installer backend in this case (i.e., providing conda/uv/whatever in the Flatpak environment) but that could be put into the `install-requirements.sh` script. Android seems like potentially the most restrictive because Chaquopy exposes a custom pip interface rather than allowing any old package installer command running? I'd worry that taking the approach necessary for Android for everything else potentially limits the possibilities for targets that don't have the same limitations.

Reply:
Agreed that it would be desirable to use the same format for both Flatpak and Android (if only because the code can be shared); and that the other benefits you describe would also be useful.

My only hesitation is whether there's an easy way to manage this in the Flatpak template itself. I'm not certain that subshell invocations (like `$(cat somefile.txt)` ) are legal in a Flatpak manifest. Definitely happy to be proven wrong on this front, though.

Reply:
It's kind of ugly, but this works as a build command in Flatpak manifest to read the arguments from a newline delimited `pip-options.txt`:

```yaml
      - readarray -t EXTRA_PIP_ARGS < pip-options.txt; CC="gcc -pthread" /app/bin/python3 -m pip install --no-cache-dir -r requirements.txt --target /app/briefcase/app_packages "${EXTRA_PIP_ARGS[@]}"
```

I don't really know if this is portable. `$SHELL` is reported as `/bin/sh` but on my Fedora installation that's just bash. I'm concerned this won't work if the build host doesn't use bash for `sh` and adding a dependency on it doesn't seem right.

I suppose this argument reading and command preparation could be written as a Python script executed with `os.execv` (something along those lines), removing the build host's shell environment as a factor altogether.

Reply:
A python script sounds like a reasonable approach to me - we definitely don't have any guarantees that the end-user will be using bash.

Reply:
I've opened a PR with a potential approach for using this options-file approach for Flatpak: https://github.com/beeware/briefcase-linux-flatpak-template/pull/53

Reply:
> Implementing another installer backend then gets split. For most targets, it would just be a change to Briefcase to support it. For targets that use the `pip-options` file, they would need to have logic in the template to use the correct installer backend, adding complexity to the template.

On Flatpak, I think the main reason we run pip inside a container rather than directly from Briefcase is to support building binary packages from source. But that wouldn't be a factor with Conda packages, which are always binary. 

So, although I haven't thought through the details, I suspect that when we add Conda support to Briefcase, we'll be running Conda directly from Briefcase on all desktop platforms including Flatpak. This will inevitably complicate the template, but I don't think anything in this PR will make it any worse.

Mobile platforms aren't currently supported by Conda at all, so we don't need to worry about that until much later, and maybe never.